### PR TITLE
impl clone for iter/values/keys of storage types

### PR DESCRIPTION
**Pull request by g4titanx**
issue no: #1121

### CODE REVIEW

File: near-sdk/src/store/free_list/iter.rs
Comment (frol) [inline near-sdk/src/store/free_list/iter.rs:92]: Wouldn't `#[derive(Clone)]` be enough? Are we absolutely need a manual implementation?

**Comment by frol** [comment]
CI still fails, by the way

File: near-sdk/src/store/free_list/iter.rs
Comment (g4titanx) [reply near-sdk/src/store/free_list/iter.rs:92]: yes, it is enough, that would work and would be more idiomatic.
i just thought, since most of the implementations were done manually, maybe that could be manual to. but that's unnecessary code i guess
i would change it to `#[derive(clone)]`

**Comment by frol** [comment]
@g4titanx Please also add the test from the original issue to demonstrate the `cycle()` now works as expected.

**Comment by frol** [comment]
@g4titanx Could you, please, address the CI errors?

**Comment by g4titanx** [comment]
> @g4titanx Could you, please, address the CI errors?

done

**Review by frol — APPROVED** [review]
Short and sweet!

@g4titanx Thank you for contributing to NEAR ecosystem!

@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
@frol Thank you for calling!
    
@g4titanx Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/g4titanx)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called
</details>

**Pull request merged**
